lib/html_parser.dart

```diff
@@ -766,20 +774,20 @@ class StyledText extends StatelessWidget {
   final InlineSpan textSpan;
   final Style style;
   final double textScaleFactor;
+  final RenderContext renderContext;
 
   const StyledText({
     this.textSpan,
     this.style,
     this.textScaleFactor = 1.0,
+    this.renderContext,
   });
 
   @override
   Widget build(BuildContext context) {
     return SizedBox(
-      width:
-          style.display == Display.BLOCK || style.display == Display.LIST_ITEM
-              ? double.infinity
-              : null,
+      width: (style.display == Display.BLOCK || style.display == Display.LIST_ITEM) && renderContext.parser.shrinkWrap != true
```

Comment:
Good lesson with boolean expressions to be had here:

```dart
renderContext.parser.shrinkWrap != true ? <expression>
```

 is the same as:
```dart
!renderContext.parser.shinkWrap ? <expression>
```

If you ever find yourself comparing to a boolean, you probably are thinking too hard and the expression can be simplified.